# Sensor Coverage Planner Ground 
sub_start_exploration_topic_ : /start_exploration
sub_terrain_map_topic_ : /terrain_map
sub_terrain_map_ext_topic_ : /terrain_map_ext 
sub_state_estimation_topic_ : /state_estimation_at_scan 
sub_registered_scan_topic_ : /registered_scan 
sub_coverage_boundary_topic_ : /sensor_coverage_planner/coverage_boundary
sub_viewpoint_boundary_topic_ : /sensor_coverage_planner/viewpoint_boundary
sub_nogo_boundary_topic_ : /sensor_coverage_planner/nogo_boundary
pub_exploration_finish_topic_ : exploration_finish
pub_runtime_breakdown_topic_ : runtime_breakdown
pub_runtime_topic_ : /runtime
pub_waypoint_topic_ : /way_point

kAutoStart : true 
kRushHome : true 
kUseTerrainHeight : false 
kCheckTerrainCollision : true 
kCheckRegisteredCloudCollision : false 
kExtendWayPoint : true 
kUseLineOfSightLookAheadPoint : true
kExtendWayPointDistance : 8.0
kKeyposeCloudDwzFilterLeafSize : 0.2
kRushHomeDist : 10
kTerrainCollisionThreshold : 0.5
kLookAheadDistance : 8
kSkipOdometryDistance : 1.5

# PlanningEnv
kUseFrontier : true 
kFrontierClusterTolerance : 1.0
kFrontierClusterMinSize : 30
kOccupancyGridOriginX: -70
kOccupancyGridOriginY: -80
kOccupancyGridOriginZ: -1 
kOccupancyGridSizeX : 500
kOccupancyGridSizeY : 500
kOccupancyGridSizeZ : 10
kOccupancyGridResolutionX : 0.3
kOccupancyGridResolutionY : 0.3
kOccupancyGridResolutionZ : 0.3
kExtractFrontierRangeX : 30
kExtractFrontierRangeY : 30
kExtractFrontierRangeZ : 2
kElminateFrontierWithLastKeypose : false 

kStackedCloudDwzLeafSize : 0.3
kPlannerCloudDwzLeafSize : 0.4
kCollisionCloudDwzLeafSize : 0.2
kCollisionCheckRadius : 0.4
kCollisionCheckPointNumThr : 1
kKeyposeCloudStackNum : 5
kCoverageZMax : 25 
kCoverageZMin : 8.0
kPointCloudRowNum : 50
kPointCloudColNum : 50
kMaxCellPointNum : 100000
kPointCloudCellSize : 20.0
kPointCloudManagerNeighborCellNum : 5
kCoverCloudZSqueezeRatio : 2.0

# ViewPointManager
viewpoint_manager/size_x : 40 
viewpoint_manager/size_y : 40 
viewpoint_manager/size_z : 1 
viewpoint_manager/rollover_step_size_x : 8
viewpoint_manager/rollover_step_size_y : 8 
viewpoint_manager/rollover_step_size_z : 8 
viewpoint_manager/resol_x : 1.2
viewpoint_manager/resol_y : 1.2
viewpoint_manager/resol_z : 1.2
kCollisionCheckTerrainThr : 0.25
kViewPointZMax : 25 
kViewPointZMin : 0 
kGreedyViewPointSampleRange : 3
kLocalPathOptimizationItrMax : 10
kCollisionRadius : 0.8
kCollisionGridZScale : 1.0 
kCollisionGridResolutionX : 0.2
kCollisionGridResolutionY : 0.2
kCollisionGridResolutionZ : 0.0
kCollisionPointThr : 1
kLineOfSightStopAtNearestObstacle : true 
kViewPointHeightFromTerrain : 0.75
kViewPointHeightFromTerrainChangeThreshold : 0.6
kCheckDynamicObstacleCollision : false
kCollisionFrameCountMax : 3

kSensorRange : 15.0
kNeighborRange : 3.0
kCoverageOcclusionThr : 0.3
kCoverageDilationRadius : 1.0

# Grid World
kGridWorldXNum : 121
kGridWorldYNum : 121
kGridWorldZNum : 121
kGridWorldCellSize : 9.6
kGridWorldCellHeight : 3.0
kGridWorldNearbyGridNum : 5
kMinAddPointNumSmall : 60 
kMinAddPointNumBig : 80
kMinAddFrontierPointNum : 50
kCellExploringToCoveredThr : 1
kCellCoveredToExploringThr: 10
kCellExploringToAlmostCoveredThr: 10 
kCellAlmostCoveredToExploringThr: 20
kCellUnknownToExploringThr: 1

# Visualization
kExploringSubspaceMarkerColorGradientAlpha : true
kExploringSubspaceMarkerColorMaxAlpha : 0.8
kExploringSubspaceMarkerColorR : 0.0
kExploringSubspaceMarkerColorG : 1.0
kExploringSubspaceMarkerColorB : 0.0
kExploringSubspaceMarkerColorA : 1.0
kLocalPlanningHorizonMarkerColorR : 0.0
kLocalPlanningHorizonMarkerColorG : 1.0
kLocalPlanningHorizonMarkerColorB : 0.0
kLocalPlanningHorizonMarkerColorA : 1.0
kLocalPlanningHorizonMarkerWidth : 0.3
kLocalPlanningHorizonHeight : 3.0